Importing distro files from Flatpak and Arch Linux

cengique opened the pull request:
Imported packaging files and new screenshots @AsciiWolf

AsciiWolf commented on misc/io.github.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2017 Daniel Rusek <mail@asciiwolf.com> -->
+<component type="desktop-application">
+  <id>com.grangerhub.Tremulous</id>
```
The app id is incorrect. It should be `io.github.Tremulous`. Why didn't you use the latest file from my [Flathub PR](https://github.com/flathub/flathub/pull/5290/files)?

cengique replied:
> The app id is incorrect. It should be `io.github.Tremulous`. Why didn't you use the latest file from my [Flathub PR](https://github.com/flathub/flathub/pull/5290/files)?

Oopsie! Pushed the correct files, but I also want to update the screenshots.

AsciiWolf commented on misc/io.github.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,61 @@
+  <provides>
+    <id>tremulous.desktop</id>
+  </provides>
+  <launchable type="desktop-id">com.grangerhub.Tremulous.desktop</launchable>
```
The launchable needs to match the desktop file name. So either rename the desktop file name or change it to the actual desktop file name.

AsciiWolf commented on misc/io.github.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,61 @@
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
+  <releases>
+    <release version="1.3.0-alpha.0.14" date="2019-09-14"/>
+    <release version="1.3.0-alpha.0.13" date="2018-02-23"/>
+    <release version="1.3.0-alpha.0.12" date="2017-10-19"/>
+  </releases>
```
Please note that the releases section is useful only if it is actively updated before every GrangerHub Tremulous release. If it is outdated, it is useless and problematic, and is better to remove it - the Flathub AppStream parser is as far as I know the only one that requires the releases section (and it can be added as a downstream patch).

cengique replied:
Removed.

AsciiWolf commented on misc/tremulous-grangerhub.desktop:
```diff
@@ -0,0 +1,8 @@
+Type=Application
+Name=GrangerHub Tremulous
+Comment=A free team based FPS/RTS hybrid built on the ioq3 engine. Includes GrangerHub community updates.
+Icon=/usr/share/pixmaps/tremulous.xpm
```
This is wrong.

1. The icon should not have a hardcoded path. It should ideally be just "tremulous", "tremulous-grangerhub" or "io.github.grangerhub.Tremulous".
2. The icon format should be png or svg.
3. The icon has to be installed in a standard location: "$PREFIX/share/icons/hicolor/*size*/apps" (replace *size* with the actual size, e.g. 512x512) or "/usr/share/icons/hicolor/scalable/apps" for a svg one.

AsciiWolf replied:
Also, again - why not using the desktop file from my Flatpak?

AsciiWolf replied:
By the way (but this is not a requirement), the icon name should ideally be the same as the desktop file name.

AsciiWolf commented on misc/tremulous-grangerhub.desktop:
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Name=GrangerHub Tremulous
```
I am not sure whether the branding is necessary. Just "Tremulous" would be better in my opinion. But it is up to you. The branding can still be in the desktop icon and it is sufficient in my opinion.

AsciiWolf commented on misc/tremulous-grangerhub.desktop:
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Name=GrangerHub Tremulous
+Comment=A free team based FPS/RTS hybrid built on the ioq3 engine. Includes GrangerHub community updates.
```
```suggestion
Comment=A free team based FPS/RTS hybrid built on the ioq3 engine. Includes GrangerHub community updates
```

Just like with the AppStream metadata summary, there should not be a dot at the end.

AsciiWolf replied:
Also, it should ideally match the AppStream summary text.

AsciiWolf commented on misc/io.github.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,61 @@
+      RTS.
+    </p>
+  </description>
+  <url type="homepage">https://grangerhub.com</url>
```
Wrong domain. ;-)

AsciiWolf commented on misc/screenshots/a-fedora.png:
The screenshots are too small by today's standards. They are probably fine for most AppStream parsers, but we should still replace them. Also, quantity does not equal quality. Two or three screenshots of a good quality are better than multiple low-quality ones.

cengique replied:
I chose 6 of them to appear in the appdata file, however, we won't be able to test until we merge this PR.

AsciiWolf commented on misc/io.github.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,61 @@
+    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/e.png</image></screenshot>
+  </screenshots>
+  <update_contact>mail@asciiwolf.com</update_contact>
+  <content_rating type="oars-1.0">
+    <content_attribute id="violence-cartoon">none</content_attribute>
+    <content_attribute id="violence-fantasy">moderate</content_attribute>
+    <content_attribute id="violence-realistic">moderate</content_attribute>
+    <content_attribute id="violence-bloodshed">mild</content_attribute>
+    <content_attribute id="violence-sexual">none</content_attribute>
+    <content_attribute id="drugs-alcohol">none</content_attribute>
+    <content_attribute id="drugs-narcotics">none</content_attribute>
+    <content_attribute id="drugs-tobacco">none</content_attribute>
+    <content_attribute id="sex-nudity">none</content_attribute>
+    <content_attribute id="sex-themes">none</content_attribute>
+    <content_attribute id="language-profanity">none</content_attribute>
+    <content_attribute id="language-humor">none</content_attribute>
+    <content_attribute id="language-discrimination">none</content_attribute>
+    <content_attribute id="social-chat">intense</content_attribute>
+    <content_attribute id="social-info">none</content_attribute>
+    <content_attribute id="social-audio">none</content_attribute>
+    <content_attribute id="social-location">none</content_attribute>
+    <content_attribute id="social-contacts">none</content_attribute>
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
```
Please replace this with the oars 1.1 rating from my PR.

AsciiWolf commented:
Looks much better now, thanks! :-)

AsciiWolf commented on misc/grangerhub-logo.png:
The icon file should be renamed to io.github.grangerhub.Tremulous.png.

cengique commented on misc/grangerhub-logo.png:
I added the Grangerhub icon for other purposes and I was thinking of using the old 48x48 dretch icon, but yours is a bit nicer, so I brought it back. :)

cengique commented:
@AsciiWolf just pushed some more commits, please review. Thanks! :)

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -19,16 +19,47 @@
-    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/d.png</image></screenshot>
-    <screenshot><image>https://raw.githubusercontent.com/hughsie/fedora-appstream/master/screenshots-extra/tremulous/e.png</image></screenshot>
+    <screenshot type="default">
+      <image type="source" width="1280" height="960">https://github.com/GrangerHub/tremulous/blob/master/misc/grangerhub-trem1.3-7d1b.jpg?raw=true</image>
```
Just a note: I may be wrong, but I think that the correct URL should be `https://raw.githubusercontent.com/GrangerHub/tremulous/master/misc/grangerhub-trem1.3-7d1b.jpg`.

Same with the other screenshots/videos.

cengique replied:
Yeah, this sounds right. I used to be able to get those URLs but somehow I can't seem to get to them. But just using your example works (for existing files). I will re-confirm once I merge this PR to master.

AsciiWolf commented:
Nice, I think it looks good now (except for the screenshots URL)! The only remaining thing are the Makefile rules. :-)

AsciiWolf commented:
Although, we should probably sort through the screenshots, select the best ones and remove others. Currently, there are too many of them and some are very similar.

cengique commented:
> Nice, I think it looks good now (except for the screenshots URL)! The only remaining thing are the Makefile rules. :-)

Great! I will sort through the screenshots, but I will need some help with the Makefile.

@wtfbbqhax @jkent can you help crafting an install section in the Makefile that can place binaries, icon, and desktop files in the proper places? As far as I see, we only have one target in the Makefile that just produces a zip or other file based on target distro. So we never worried about an install section. What's the situation with the CMake build? Was that ever working? Thanks!

cengique commented:
Looks like libcurl is also outdated and causing Linux build errors due to `USE_RESTCLIENT=1`.

cengique commented:
@AsciiWolf @jkent I added an `install:` target in the Makefile. I'd appreciate if you can review. Note that the game client is not perfectly working in this branch, so I'll rebase to a release branch to complete testing. I'm looking for general feedback about the changes in the Makefile. Thanks!

@AsciiWolf I removed screenshot `d-fedora.png`, but kept the newer ones as not all are referenced in the flatpak.

AsciiWolf commented on Makefile:
```diff
@@ -111,28 +111,44 @@ ifndef VERSION
 
 ifndef COPYDIR
-COPYDIR="/usr/local/games/tremulous"
+COPYDIR="/opt/$(PACKAGE)"
```
Why `/opt`?

cengique replied:
I was following what was done in the Arch Linux packaging. Since Tremulous binaries need to stay in the data folder, it doesn't correspond to the usual Linux way of storing stuff under PREFIX/share/APPNAME format. But I'm not confident in that. There used to be a separate tremulous-data package. Not sure which approach is better.

cengique replied:
Also, /usr/local/games seems to be an unused location in modern distros. Again, I'm not confident about this.

AsciiWolf replied:
But /opt is usually not used anymore by most software. You can use something like `$PREFIX/share/tremulous` (/usr/local/share/tremulous if PREFIX is /usr/local).

Take a look how we package it in Fedora:
https://koji.fedoraproject.org/koji/rpminfo?fileOrder=name&rpmID=37498623&buildrootOrder=-id&buildrootStart=0#filelist
https://koji.fedoraproject.org/koji/rpminfo?fileOrder=name&rpmID=37497768&buildrootOrder=-id&buildrootStart=0#filelist

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,46 @@ distclean: clean toolsclean
+# Install the .desktop, icon files, license, etc.
+	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.png" "$(PREFIX)/share/pixmaps/tremulous.png"
+	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.desktop" \
+			 "$(PREFIX)/share/applications/tremulous.desktop"
```
No, this is incorrect. The whole point of renaming the desktop file (and its matching launchable tag in the AppStream metadata file) was to keep it at the new `io.github.grangerhub.Tremulous.desktop` name. Installing it as `tremulous.desktop` makes no sense at this point and will break things for AppStream metadata.

If you want to install the desktop/icon files as just `tremulous.*`, rename the original files in the misc directory and also change the launchable tag in AppStream file to just "tremulous.desktop" (but don't change the actual id and metainfo file name - that should always stay in rDNS format).

cengique replied:
Got it. I renamed them to Tremulous-Grangerhub.

AsciiWolf replied:
No, this is still incorrect and even more broken now! I was talking about the rDNS formatted name and I even sent suggestions to apply.

AsciiWolf replied:
By the way, rDNS naming conventions for desktop files will probably became compulsory on many Linux distributions / packaging formats in the future. So my change was also future-proof.

If you use this "Tremulous-Grangerhub" name format because of the upcoming official Grangerhub Snap, then please consider doing these changes in downstream (patch for the Snap package), not directly in upstream.

But it is up to you. I just do not personally like these changes, that's probably my problem.

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,46 @@ distclean: clean toolsclean
+	@$(INSTALL) -d $(PREFIX)/bin
+	@$(INSTALL) -D -m755 $(BR)/$(CLIENTBINSH) $(BR)/$(SERVERBINSH) $(PREFIX)/bin
+# Install the .desktop, icon files, license, etc.
+	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.png" "$(PREFIX)/share/pixmaps/tremulous.png"
```
I am not sure I understand. The icon name in the desktop file is `io.github.grangerhub.Tremulous` so it should be installed as `io.github.grangerhub.Tremulous.png`, not `tremulous.png`.

Also, please never use the pixmaps directory, it is deprecated and already was for last 15 years or so. Use `$(PREFIX)/share/icons/hicolor/128x128/apps`.

cengique replied:
Also renamed icon to Tremulous-Grangerhub. Corrected the install directory.

AsciiWolf commented on misc/screenshots/a-fedora.png:
We can remove the actual *-fedora.png files since they are no longer used in the AppStream metadata file.

cengique replied:
True, they are redundant with the better ones anyway.

AsciiWolf commented:
It looks good except the things that I pointed out. Thanks!

AsciiWolf commented:
By the way, we have a CI build failure on Linux:

`error: 'CURLMOPT_MAX_TOTAL_CONNECTIONS' conflicts with a previous declaration`

Is it the same thing as already mentioned [here](https://github.com/GrangerHub/tremulous/pull/130#issuecomment-2147954605)?

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,46 @@ distclean: clean toolsclean
+	@$(INSTALL) -d $(PREFIX)/bin
+	@$(INSTALL) -D -m755 $(BR)/$(CLIENTBINSH) $(BR)/$(SERVERBINSH) $(PREFIX)/bin
+# Install the .desktop, icon files, license, etc.
+	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.png" "$(PREFIX)/share/pixmaps/tremulous.png"
```
```suggestion
	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.png" "$(PREFIX)/share/icons/hicolor/128x128/apps/io.github.grangerhub.Tremulous.png"
```

cengique replied:
I think I already fixed these. Didn't know you could make suggestions here!

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,46 @@ distclean: clean toolsclean
+# Install the .desktop, icon files, license, etc.
+	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.png" "$(PREFIX)/share/pixmaps/tremulous.png"
+	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.desktop" \
+			 "$(PREFIX)/share/applications/tremulous.desktop"
```
```suggestion
			 "$(PREFIX)/share/applications/io.github.grangerhub.Tremulous.desktop"
```

cengique replied:
also fixed

cengique commented:
> By the way, we have a CI build failure on Linux:
> 
> `error: 'CURLMOPT_MAX_TOTAL_CONNECTIONS' conflicts with a previous declaration`
> 
> Is it the same thing as already mentioned [here](https://github.com/GrangerHub/tremulous/pull/130#issuecomment-2147954605)?

That's right, we will work on that separately, but for now you can build with:
```
USE_INTERNAL_LUA=1 make -j
```
by removing the building of the REST client.

cengique commented:
@AsciiWolf I think I addressed all the issues. Please check! Thanks :)

AsciiWolf commented on Makefile:
```diff
@@ -2947,27 +2947,24 @@ $(B)/$(SERVERBINSH):
-			 "$(PREFIX)/share/applications/tremulous.desktop"
-	@$(INSTALL) -d "$(PREFIX)/share/metainfo/" "$(PREFIX)/share/licenses/$(PACKAGE)/"
-	@$(INSTALL) -D -m644 "misc/io.github.grangerhub.Tremulous.appdata.xml" \
+	$(INSTALL) -D -m755 $(BR)/$(CLIENTBINSH) $(PREFIX)/bin/tremulous-grangerhub
```
To be honest, I really dislike this branding thing. Just "tremulous" and "tremded" was fine and there is a very little chance that user will have both grangerhub and non-grangerhub Tremulous installations on their system, in the same paths.

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -7,8 +7,8 @@
-  <launchable type="desktop-id">io.github.grangerhub.Tremulous.desktop</launchable>
-  <name>Tremulous</name>
+  <launchable type="desktop-id">Tremulous-Grangerhub.desktop</launchable>
+  <name>Tremulous-Grangerhub</name>
```
Why? This will only make it confusing for our users. You can instead mention in the description that this is a "Grangerhub version of Tremulous". And it will add me much work to do to adjust the Flatpak to this unnecessary change. :-/

AsciiWolf commented on misc/Tremulous-Grangerhub.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Tremulous
+Comment=Aliens vs Humans, First Person Shooter game with elements of Real Time Strategy
+Icon=io.github.grangerhub.Tremulous
```
If you really want to use the "Tremulous-Grangerhub" naming scheme (which I strongly disagree with as I mentioned), you need to fix the Icon name here as well.

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,43 @@ distclean: clean toolsclean
+install: release
+ifneq (,$(findstring "$(PLATFORM)", "linux" "gnu_kfreebsd" "kfreebsd-gnu" "gnu"))
+	$(echo_cmd) "Installing for Linux platform in $(COPYBINDIR) and $(PREFIX)"
+	@$(INSTALL) -d $(PREFIX)/bin "$(PREFIX)/share/metainfo/" \
```
Just a very small nitpick here. I noticed that some paths have "/" at their end and some don't. It does not matter, but it would still be nice to make it consistent.

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,71 @@
+      Players on both teams are able to build working structures in-game like an
+      RTS. 
+    </p>
+    <p>
+      This snap provides the Grangerhub updated client binary that
+      supports new features.
+    </p>
```
```suggestion
```

The snap related (downstream) information should not be in upstream metadata.

AsciiWolf commented:
So, I have thought about the desktop/icon file name one more time and I come to conclusion that **you can probably keep the name at tremulous-grangerhub if you want**, but please change the name to a lower case one (tremulous-grangerhub.desktop / tremulous-grangerhub.png). Most desktop files that are in legacy non-rDNS name format have their whole name in lowercase and also, it looks better that way and is more consistent with the original (tremulous.desktop / tremulous.png) name. And don't forget to update the _Icon_ entry in the desktop file to use the correct icon.

But please, consider changing the `<name>` tag in AppStream metadata from `<name>Tremulous-Grangerhub</name>` back to `<name>Tremulous</name>` and adding an information about the GrangerHub Tremulous version to a `<description>` instead.

Thanks!

cengique commented:
> So, I have thought about the desktop/icon file name one more time and I come to conclusion that **you can probably keep the name at tremulous-grangerhub if you want**, but please change the name to a lower case one (`tremulous-grangerhub.desktop` / `tremulous-grangerhub.png`). Most desktop files that are in legacy non-rDNS name format have their whole name in lowercase, and it also looks better that way and is more consistent with the original (tremulous.desktop / tremulous.png) name. And don't forget to update the _Icon_ entry in the desktop file to use the correct icon.
> 
> But please, consider changing the `<name>` tag in AppStream metadata from `<name>Tremulous-Grangerhub</name>` back to `<name>Tremulous</name>` and adding an information about the GrangerHub Tremulous version to a `<description>` instead.
> 
> Thanks!

Hey @AsciiWolf , thanks for the compromising solution. I was a bit taken aback and wanted a meeting since this was taking too long. We can still have a short meeting to find the best solutions and finalize this. I have no problem changing the name tag in AppStream back to Tremulous. The reason for branding was that there are a couple of different clients around (the other one being Tremfusion) - but I guess we can name that one separately if it comes to it. Also the GrangerHub versions may be buggy and people may want to install vanilla versions (if they can find working ones!). But the GrangerHub info can stay in the description and version fields for now. I will change the files to lowercase as I noticed that it looked awkward in those folders, which mostly contained lowercase tools. Although, I saw a couple of font file names, etc, that had mixed case. Anyways, I'll think about the naming scheme one more time and will go through the list of items you picked above. Thanks.

AsciiWolf commented:
Thanks for your reply! Meeting could be a good idea, but I am afraid that my time schedule does not allow me more meetings in upcoming week or two. :-/

Regarding the branding name: Most Linux distributions do not package Tremfusion or vanilla Tremulous anymore, so I would not worry about that. :-)

cengique commented:
> So, I have thought about the desktop/icon file name one more time and I come to conclusion that **you can probably keep the name at tremulous-grangerhub if you want**, but please change the name to a lower case one (`tremulous-grangerhub.desktop` / `tremulous-grangerhub.png`). Most desktop files that are in legacy non-rDNS name format have their whole name in lowercase, and it also looks better that way and is more consistent with the original (tremulous.desktop / tremulous.png) name. And don't forget to update the _Icon_ entry in the desktop file to use the correct icon.
> 
> But please, consider changing the `<name>` tag in AppStream metadata from `<name>Tremulous-Grangerhub</name>` back to `<name>Tremulous</name>` and adding an information about the GrangerHub Tremulous version to a `<description>` instead.

@AsciiWolf I think I fixed these, but please take a look at my latest commit. Thanks and sorry for the long delay. Too much traveling and not enough time on the computer.

AsciiWolf commented on misc/tremulous-grangerhub.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Tremulous Grangerhub
+Comment=Aliens vs Humans, First Person Shooter game with elements of Real Time Strategy (Grangerhub version)
+Icon=tremulous-grangerhub
```
We can change the icon file name to io.github.grangerhub.Tremulous.png instead. Same with the desktop file name (io.github.grangerhub.Tremulous.desktop).

AsciiWolf commented on misc/tremulous-grangerhub.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Tremulous Grangerhub
```
Is the " Grangerhub" suffix in name really necessary if we already have it in the Comment section (and also in the Summary section of AppStream metadata file)?

We could optionally modify the icon file and add something like "GH" text to the corner to make the branding more apparent. But I do not think that changing the name of the actual game is a good idea. (Especially considering that the Grangerhub version is de facto the only version of Tremulous that is still used nowadays.)

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,67 @@
+  <provides>
+    <id>tremulous-grangerhub.desktop</id>
+  </provides>
+  <launchable type="desktop-id">Tremulous-Grangerhub.desktop</launchable>
```
When/if the desktop file name is changed, we have to change it here (and also in the provides section) as well. Also, please note that the file name is case sensitive, so the current entry is incorrect.

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,43 @@ distclean: clean toolsclean
+	$(INSTALL) -D -m755 $(BR)/$(CLIENTBINSH) $(PREFIX)/bin/tremulous-grangerhub
+	$(INSTALL) -D -m755 $(BR)/$(SERVERBINSH) $(PREFIX)/bin/tremded-grangerhub
+	$(INSTALL) -D -m644 "misc/Tremulous-Grangerhub.png" "$(PREFIX)/share/icons/hicolor/128x128/apps/"
+	$(INSTALL) -D -m644 "misc/Tremulous-Grangerhub.desktop" \
```
Wait... Why are we installing the file with different (upper case) name? It does not make sense.

AsciiWolf commented on Makefile:
```diff
@@ -2904,6 +2931,43 @@ distclean: clean toolsclean
+	@cd $(BR) && for file in $(NAKED_TARGETS); do \
+		$(INSTALL) -D $$file $(COPYBINDIR)/$$file; \
+	done
+	$(INSTALL) -D -m755 $(BR)/$(CLIENTBINSH) $(PREFIX)/bin/tremulous-grangerhub
```
We are still installing the tremulous and tremded binaries with -grangerhub suffixes. Is this intended?

AsciiWolf commented:
Thanks! I have provided some feedback.

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,67 @@
+  <launchable type="desktop-id">Tremulous-Grangerhub.desktop</launchable>
+  <name>Tremulous</name>
+  <developer_name>Dark Legion Development and GrangerHub</developer_name>
+  <summary>Aliens vs Humans, First Person Shooter game with elements of Real Time Strategy (Grangerhub version)</summary>
```
Just a question: Is the correct name _Granger**h**ub_ or _Granger**H**ub_?

cengique commented:
@AsciiWolf just got around to fixing these issues. If you can check the conversations and resolve, we may be close to being done! If this is all good, I'll rebase this branch to a working release and do more testing.

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.desktop:
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Tremulous
+Comment=Aliens vs Humans, First Person Shooter game with elements of Real Time Strategy (GrangerHub version)
+Icon=io.github.grangerhub.Tremulous.png
```
```suggestion
Icon=io.github.grangerhub.Tremulous
```

The desktop icon should not contain a file type suffix.

AsciiWolf commented on Makefile:
```diff
@@ -111,22 +111,38 @@ ifndef VERSION
 BASEGAME_CFLAGS=-I../../${MOUNT_DIR}
 
+ifndef PREFIX
+PREFIX=/usr
```
The default prefix should probably rather be `/usr/local`.

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,67 @@
+      Players can choose from 2 unique races, aliens and humans.
+      Players on both teams are able to build working structures in-game like an
+      RTS. 
+    </p>
```
Now that the GrangerHub branding was removed from the Tremulous Flatpak/desktop file name, we could instead mention it in the description. :-)

AsciiWolf replied:
Ah, I see that it is mentioned in the summary. Maybe we could provide more information what the Grangerhub version is in the description?

AsciiWolf commented:
Looks very good now, thanks! :-)

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,67 @@
+    </p>
+  </description>
+  <url type="homepage">https://github.com/GrangerHub/tremulous</url>
+  <screenshots>
```
One last small thing: We should consider adding more screenshots with aliens. Currently, most of them are with humans. :)

cengique replied:
Added a bunch! About to commit.

AsciiWolf replied:
Thanks! We should probably also focus more on quality than quantity and keep only the good screenshot + remove screenshots that have different resolution and look inconsistent.

cengique replied:
> Thanks! We should probably also focus more on quality than quantity and keep only the good screenshots + remove screenshots that have different resolution and look inconsistent.

I already removed the low res screenshots from the appdata file. I have some extra screenshots left in the folder, but I'd like to keep them for posterity. They are not referenced from the appdata file.

cengique commented:
> Any progress here?

Thanks for the reminder @AsciiWolf ! I needed it. :) 

I added a number of alien screenshots, in addition to fixing the other issues. Thanks as usual for checking and providing feedback.

Once we're done here, I think I need to update so that we can build the flatpak for 0.15. Any other next steps?

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,117 @@
+    <p>
+      The GrangerHub version contains the updated game server and
+      client binaries with several stability improvements and new
+      features. See the GrangerHub Github page for more details:
+      https://github.com/GrangerHub/tremulous/
```
```suggestion
      features. See the homepage for more details.
```

(Actual URLs with http(s) are not supported inside the description. Also, this URL is already in the homepage tag that gets displayed in AppStream UIs as the project homepage.)

AsciiWolf replied:
Last thing. :)

cengique commented:
@AsciiWolf removed URL from appdata as requested. I posted a comment [above](#discussion_r1921649603) regarding screenshots.

I also corrected the Github Actions for the Linux build, which may help with fixing the issues while building the Flatpak.

AsciiWolf commented:
Thanks! LGTM. :-)

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,116 @@
+  <launchable type="desktop-id">io.github.grangerhub.Tremulous.desktop</launchable>
+  <name>Tremulous</name>
+  <developer_name>Dark Legion Development and GrangerHub</developer_name>
+  <summary>Aliens vs Humans, First Person Shooter game with elements of Real Time Strategy (Grangerhub version)</summary>
```
```suggestion
  <summary>Aliens vs Humans, First Person Shooter game with elements of Real Time Strategy (GrangerHub version)</summary>
```

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2017 Daniel Rusek <mail@asciiwolf.com> -->
```
Feel free to remove my copyright. :-)

AsciiWolf commented on .github/workflows/build-macos.yml:
```diff
@@ -1,4 +1,4 @@
-name: Build MacOS 11
+name: Build for MacOS
```
```suggestion
name: Build for macOS
```

It's lowercase macOS nowadays. :)

AsciiWolf commented on misc/io.github.grangerhub.Tremulous.appdata.xml:
```diff
@@ -0,0 +1,116 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2017-2025 Daniel Rusek <mail@asciiwolf.com> and Cengiz Gunay <cengique@users.sf.net> -->
```
Looks good. :)
